kobo/apps/trash_bin/tests/test_utils.py

```diff
@@ -331,3 +341,174 @@ def test_delete_project(self):
             )
             == 0
         )
+
+    def test_attachment_storage_updates_on_project_trash_and_restore(self):
+        """
+        Ensure that `attachment_storage_bytes` in UserProfile and XForm is
+        updated correctly when a project is moved to the trash and restored
+        """
+        user = User.objects.get(username='someuser')
+        asset = Asset.objects.create(
+            asset_type='survey',
+            content={
+                'survey': [
+                    {'type': 'audio', 'label': 'q1', 'name': 'q1'},
+                    {'type': 'file', 'label': 'q2', 'name': 'q2'}
+                ]
+            },
+            owner=user
+        )
+        asset.save()
+        asset.deploy(backend='mock')
+
+        username = user.username
+        submission = {
+            'q1': 'audio_conversion_test_clip.3gp',
+            'q2': 'audio_conversion_test_image.jpg',
+            '_uuid': str(uuid.uuid4()),
+            '_attachments': [
+                {
+                    'download_url': f'http://testserver/{username}/audio_conversion_test_clip.3gp',  # noqa: E501
+                    'filename': f'{username}/audio_conversion_test_clip.3gp',
+                    'mimetype': 'video/3gpp',
+                },
+                {
+                    'download_url': f'http://testserver/{username}/audio_conversion_test_image.jpg',  # noqa: E501
+                    'filename': f'{username}/audio_conversion_test_image.jpg',
+                    'mimetype': 'image/jpeg',
+                },
+            ],
+            '_submitted_by': username,
+        }
+        asset.deployment.mock_submissions([submission])
+
+        # Fetch the storage data before moving the project to trash
+        xform_before = XForm.all_objects.filter(kpi_asset_uid=asset.uid).aggregate(
+            total_bytes=Sum('attachment_storage_bytes')
+        )['total_bytes']
+        user_profile_before = UserProfile.objects.filter(user_id=user.pk).aggregate(
+            total_bytes=Sum('attachment_storage_bytes')
+        )['total_bytes']
+
+        # Move project to trash
+        ProjectTrash.toggle_statuses(
+            [asset.uid], active=False, toggle_delete=True
+        )
```

Comment:
You need to add a test when projects are only archived

Reply:
Please split this test: one for deletion, one for archiving (even if they use almost the same code).